File: HackerRank Challenge Solutions/Day 9/Solution.java
  import java.io.*;
  import java.util.*;
  import java.text.*;
  import java.math.*;
  import java.util.regex.*;

  public class Solution {
  
      public static void main(String args[]){
          Scanner sc=new Scanner(System.in);
          //Take Input
          int a = sc.nextInt();
          int b = sc.nextInt();
          int gcd=find_gcd(a,b);
          System.out.println(gcd);
      }
      static  int find_gcd(int a,int b){
          //Write the base condition
          if (a == b || b == 0) {
              return a;
          } else {
              return find_gcd(b,a%b);
          }
      }
  }

File: HackerRank Challenge Solutions/Day 9/Solution.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
int find_gcd(int,int); // Function Prototype
int main() {
    int a,b;
    //Take input
    cin>>a>>b;
    int gcd=find_gcd(a,b); // Function Call
    cout<<gcd;
    return 0;   
}

int find_gcd(int a , int b) // Function Definition 
{
    if(a==b ||b==0){
        return a;
    }
    else
        return find_gcd(b,a%b);
}